proto/proto/store.proto

```diff
@@ -64,44 +66,232 @@ service Rpc {
+    rpc ApplyBlock(ApplyBlock) returns (google.protobuf.Empty) {}
 
     // Retrieves block header by given block number. Optionally, it also returns the MMR path
     // and current chain length to authenticate the block's inclusion.
-    rpc GetBlockHeaderByNumber(requests.GetBlockHeaderByNumberRequest) returns (responses.GetBlockHeaderByNumberResponse) {}
+    rpc GetBlockHeaderByNumber(shared.GetBlockHeaderByNumber) returns (shared.BlockHeaderByNumber) {}
 
     // Returns data required to prove the next block.
-    rpc GetBlockInputs(requests.GetBlockInputsRequest) returns (responses.GetBlockInputsResponse) {}
+    rpc GetBlockInputs(GetBlockInputs) returns (BlockInputs) {}
 
     // Returns the inputs for a transaction batch.
-    rpc GetBatchInputs(requests.GetBatchInputsRequest) returns (responses.GetBatchInputsResponse) {}
+    rpc GetBatchInputs(GetBatchInputs) returns (BatchInputs) {}
 
     // Returns data required to validate a new transaction.
-    rpc GetTransactionInputs(requests.GetTransactionInputsRequest) returns (responses.GetTransactionInputsResponse) {}
+    rpc GetTransactionInputs(GetTransactionInputs) returns (TransactionInputs) {}
 }
 
 // Store API for the network transaction builder component
 service NtxBuilder {
     // Retrieves block header by given block number. Optionally, it also returns the MMR path
     // and current chain length to authenticate the block's inclusion.
-    rpc GetBlockHeaderByNumber(requests.GetBlockHeaderByNumberRequest) returns (responses.GetBlockHeaderByNumberResponse) {}
+    rpc GetBlockHeaderByNumber(shared.GetBlockHeaderByNumber) returns (shared.BlockHeaderByNumber) {}
 
     // Returns the list of unconsumed network notes and the next page number to query.
-    rpc GetUnconsumedNetworkNotes(requests.GetUnconsumedNetworkNotesRequest) returns (responses.GetUnconsumedNetworkNotesResponse) {}
+    rpc GetUnconsumedNetworkNotes(GetUnconsumedNetworkNotes) returns (UnconsumedNetworkNotes) {}
 
     // Returns the block header at the chain tip, as well as the MMR peaks corresponding to this 
     // header for executing network transactions. 
-    rpc GetCurrentBlockchainData(requests.GetCurrentBlockchainDataRequest) returns (responses.GetCurrentBlockchainDataResponse) {}
+    rpc GetCurrentBlockchainData(GetCurrentBlockchainData) returns (CurrentBlockchainData) {}
 
     // Returns the latest state of a network account with the specified account prefix.
-    rpc GetNetworkAccountDetailsByPrefix(requests.GetNetworkAccountDetailsByPrefixRequest) returns (responses.GetNetworkAccountDetailsByPrefixResponse) {}
+    rpc GetNetworkAccountDetailsByPrefix(GetNetworkAccountDetailsByPrefix) returns (NetworkAccountDetailsByPrefix) {}
+}
+
+// Applies changes of a new block to the DB and in-memory data structures.
+message ApplyBlock {
+    // Block data encoded using [winter_utils::Serializable] implementation for
+    // [miden_objects::block::Block].
+    bytes block = 1;
+}
```

Comment:
This should probably be just called `Block` and be in the `types/blockchain.proto`.